GS/HW: Remove legacy code for changing RT from depth to color in TC invalidation.

TJnotJT opened the pull request:
### Description of Changes
1. Removed a couple lines in TC invalidation logic that converts some targets to C32.
2. Changed behavior of LookupSource() when we find a target with exact BP match but the requested area is outside the valid area of the target. In this case create a temporary source and load from memory.

### Rationale behind Changes
This removed code was incorrectly changing the format of targets in the TC. Intended to fix https://github.com/PCSX2/pcsx2/issues/8990. 

### Suggested Testing Steps
Only affects HW renderers. Testing other the game in the issue linked above and any other games would be helpful.

### Did you use AI to help find, test, or implement this issue or feature?
No.

### Credits
Co-authored by refraction.

TJnotJT committed:
GS/HW: Remove legacy code for changing RT from depth to color in TC invalidation.

Co-authored-by: refraction

JordanTheToaster reviewed (approved):
Fixes the light leak fully in TimeSplitters Future Perfect GT4s car icons look correct.

TJnotJT commented:
Dump run in Vulkan shows a regression with Ultraman_Fighting_Evolution_-_Rebirth_SLPS-25529_20230303124827.gs.xz, so keeping as a draft.

TJnotJT committed:
GS/HW/TC: Remove outside target PSM check in LookupSource.

Co-autored-by: refraction

TJnotJT committed:
GS/HW/TC: Force a temporary source creation in edges cases.

Case: When looking up a source, we find a perfect BP hit for a target.
However, the requested area is outside the target's valid area.
Don't use the target direct and instead load from memory in a temporary source.

Co-authored-by: refraction

TJnotJT commented:
Added a commit to take care of edge case where LookupSource sees an exact target BP hit but the requested area for the source is outside the target's valid area. Fixes a few regressions. Still need to do a full dump run.

TJnotJT commented:
Vulkan dump run came clean so opening for review.

JordanTheToaster reviewed (approved):
Re tested TS FP and GT4 and a few other games and all is still good.

lightningterror reviewed (commented):
This breaks dump `Radiata Stories_SLUS-21262_20230521194640.gs.xz` after the second time the dump is played back:

<img width="1643" height="545" alt="image" src="https://github.com/user-attachments/assets/513ce410-ba8d-4459-aac4-a10967d971e8" />

TJnotJT committed:
GS/HW/TC: Do not force temporary source creation for PSMT8 sources that request outside a target valid area.

TJnotJT commented:
After the lastest push, `Radiata Stories_SLUS-21262_20230521194640.gs.xz` looks the same on PR and master. Also redid a dump run on Vulkan.

Side note: the draw that caused the difference is actually bugged in both PR and master; it uses a texture that was in VRAM on the first loop but is no longer there on the 2nd loop. Now PR and master just load the same incorrect texture.

lightningterror reviewed (approved):
Nice, just wanted to make sure.

bslenul commented:
Hello! As mentioned by prafullpcsx2 in my Onimusha 3 issue (https://github.com/PCSX2/pcsx2/issues/14578#issuecomment-4683037386 for a GS dump), this PR seems to cause issues on the blobs enemy textures when using HW renderer.

This is from PCSX2 v2.5.291 (before this got merged):
<img width="1134" height="697" alt="image" src="https://github.com/user-attachments/assets/05be4a8b-0c2a-48b4-a333-c340dab347ad" />
And this is from v2.5.292 (after it got merged):
<img width="1134" height="697" alt="image" src="https://github.com/user-attachments/assets/2954b412-62ca-430d-8572-80b5a41481a4" />

TJnotJT commented:
Thank for reporting the issue, will try to take a look soon
